storm-core/src/jvm/org/apache/storm/scheduler/blacklist/BlacklistScheduler.java

```
@@ -0,0 +1,212 @@
+        }
+        if(_conf.containsKey(Config.BLACKLIST_SCHEDULER_TOLERANCE_COUNT)){
+            toleranceCount=(Integer)_conf.get(Config.BLACKLIST_SCHEDULER_TOLERANCE_COUNT);
+        }
+        if(_conf.containsKey(Config.BLACKLIST_SCHEDULER_RESUME_TIME)){
+            resumeTime=(Integer)_conf.get(Config.BLACKLIST_SCHEDULER_RESUME_TIME);
+        }
+        String reporterClassName=_conf.containsKey(Config.BLACKLIST_SCHEDULER_REPORTER)?(String)_conf.get(Config.BLACKLIST_SCHEDULER_REPORTER):"";
+        try {
+            reporter=(IReporter)Class.forName(reporterClassName).newInstance();
+        } catch (ClassNotFoundException e) {
+            LOG.error("Can't find blacklist reporter for name {}",reporterClassName);
+        } catch (InstantiationException e) {
+            LOG.error("Throw InstantiationException blacklist reporter for name {}", reporterClassName);
+        } catch (IllegalAccessException e) {
+            LOG.error("Throw illegalAccessException blacklist reporter for name {}", reporterClassName);
+        }
+
+        String strategyClassName=_conf.containsKey(Config.BLACKLIST_SCHEDULER_STRATEGY)?(String)_conf.get(Config.BLACKLIST_SCHEDULER_STRATEGY):"";
+        try {
+            blacklistStrategy=(IBlacklistStrategy)Class.forName(strategyClassName).newInstance();
+        } catch (ClassNotFoundException e) {
+            LOG.error("Can't find blacklist strategy for name {}",reporterClassName);
+        } catch (InstantiationException e) {
+            LOG.error("Throw InstantiationException blacklist strategy for name {}", reporterClassName);
+        } catch (IllegalAccessException e) {
+            LOG.error("Throw illegalAccessException blacklist strategy for name {}", reporterClassName);
+        }
+
+        nimbusMonitorFreqSecs=(Integer)_conf.get(Config.NIMBUS_MONITOR_FREQ_SECS);
+        blacklistStrategy.prepare(reporter,toleranceTime,toleranceCount,resumeTime,nimbusMonitorFreqSecs);
```

You catch errors above without handling the error, which would leave `blacklistStrategy` null. This will throw an NPE here. Probably better to just throw the exceptions when `blacklistStrategy` can't be found/instantiated.